modules/gdscript/gdscript_functions.cpp

```diff
@@ -1176,21 +1180,12 @@ void GDScriptFunctions::call(Function p_func, const Variant **p_args, int p_arg_
 
 				return;
 			}
+			// Normalize script extension (gdc, gde)
+			String ext = GDScriptLanguage::get_singleton()->get_extension();
+			String path = String(d["@path"]).get_basename() + "." + ext;
 
-			Ref<Script> scr = ResourceLoader::load(d["@path"]);
-			if (!scr.is_valid()) {
-
-				r_error.error = Variant::CallError::CALL_ERROR_INVALID_ARGUMENT;
-				r_error.argument = 0;
-				r_error.expected = Variant::OBJECT;
-				r_ret = RTR("Invalid instance dictionary format (can't load script at @path)");
-				return;
-			}
```

Comment:
This whole check seems to be redundant because we're only interested in GDScript resources here.